### ISSUE TITLE
I am still having this issue...

### ISSUE THREAD

**Issue by rawansuww** [tensorflow/tensorflow#48898]
@HaraBeron This is a know bug with `accuracy`. In earlier TF versions `accuracy` was correctly inferred from the `loss` function. In this case, `accuracy` need to be inferred as `sparse_categorical_accuracy` but in the current version it is not working as expected.

Everything works as expected If you replace `accuracy` with `sparse_categorical_accuracy` in `model.compile` as shown below.

```
model.compile(optimizer='adam',
              loss=tf.losses.SparseCategoricalCrossentropy(from_logits=True),
              metrics=['sparse_categorical_accuracy'])
```

_Originally posted by @jvishnuvardhan in https://github.com/tensorflow/tensorflow/issues/42890#issuecomment-685998646_

Can someone please help me about the above? I already trained a few models with TF=2.3.0, then when I tried loading the model to print some ROC curves, I got zero-accuracy across everything. I tried to compile with 'sparse_categorical_accuracy' as referenced above after I load_model, but it did not work.. Do I have to re-train all my models because of this bug?

Please help

**Comment by bhack** [comment]
Is this a dup of https://github.com/tensorflow/tensorflow/issues/41361?

**Comment by rawansuww** [comment]
> Is this a dup of #41361?

No, I don't think so.  It's the same issue as #42890 and was marked as closed. I tried the recommended bug fix of using 'sparse_categorical_accuracy' when I compile after load_model, but it did not work

Thanks for the fast response :)

**Comment by jvishnuvardhan** [comment]
@rawansuww Can you please share a standalone code to reproduce the issue? Thanks!
